code/modules/makeshift/makeshift_reagents.dm

```diff
@@ -0,0 +1,248 @@
+
+/obj/structure/chemkit/Initialize()
+	. = ..()
+	create_reagents(180)
+
+/obj/structure/chemkit/examine(mob/user)
+	. = ..()
+	if(analyzer)
+		to_chat(user, span("notice", "The analyzer displays that the temperature is [round(reagents.get_temperature() - T0C,0.1)]C."))
+
+/obj/structure/chemkit/verb/phase_filter()
+	set name = "Set Phase Filter"
+	set category = "Object"
+	set src in view(1)
+	if(!use_check(usr))
+		return 0
+	phase_filter = input("Which phase do you want to filter?", "Phase Filter", null) as null|anything in list("solid", "liquid", "gas", "none")
+	if(phase_filter)
+		to_chat(usr, span("notice", "You switch to a [phase_filter] filter."))
+	if(phase_filter == "none")
+		phase_filter = FALSE
+
+/obj/structure/chemkit/verb/eject_contents()
+	set name = "Remove Contents"
+	set category = "Object"
+	set src in view(1)
+	for(var/atom/movable/A in contents)
+		if(A == analyzer)
+			continue
+		A.forceMove(loc)
+
+/obj/structure/chemkit/attack_hand(mob/user)
+	transfer_out = !transfer_out
+	to_chat(user, span("notice", "You are now [transfer_out ? "removing from" : "adding to"] \the [src]."))
+
+/obj/structure/chemkit/proc/heat_item(obj/item/weapon/W, mob/user)
+	var/joules = 0
+	if(istype(W, /obj/item/weapon/flame))
+		joules = 1000 // 1 kJ per match, and we're assuming lighters give as much per use
+	else if(W.iswelder())
+		joules = 10000 // we'll just have it be ten times stronger
+	else if(istype(W, /obj/item/device/assembly/igniter))
+		joules = 5000 // half as strong as a welder; this thing has to set off bombs and such
+	else
+		joules = 500 // we're assuming it's some kind of novelty toy like lunea's gloves, etc; not exactly a match but it still works
+	reagents.add_thermal_energy(joules)
+	user.visible_message(span("warning", "[user] holds \the [W] up to \the [src]!"), span("notice", "You use \the [W] to heat \the [src]'s contents."), span("notice", "You hear something sizzle."))
+
+/obj/structure/chemkit/proc/trans_item(obj/item/I, mob/user)
+	if(transfer_out)
+		var/amt = 0
+		switch(phase_filter)
+			if("solid")
+				amt = reagents.trans_ids_to(I.reagents, reagents.get_ids_by_phase(SOLID), reagents.get_free_space())
+			if("liquid")
+				amt = reagents.trans_ids_to(I.reagents, reagents.get_ids_by_phase(LIQUID), reagents.get_free_space())
+			if("gas")
+				amt = reagents.trans_ids_to(I.reagents, reagents.get_ids_by_phase(GAS), reagents.get_free_space())
+			else
+				amt = reagents.trans_to_holder(I.reagents, reagents.get_free_space())
+		if(amt)
+			to_chat(user, span("notice", "You fill \the [I] with [amt] units from \the [src]."))
+	else if(I.reagents && I.reagents.total_volume)
+		var/amt = I.reagents.trans_to_holder(reagents, I.reagents.total_volume) // just pour it if you can
+		to_chat(user, span("notice", "You pour [amt] units from \the [I] into \the [src]."))
+
+/obj/structure/chemkit/proc/smash(obj/item/I, mob/user)
```

Comment:
Holy. Christ. While I understand that this is SS13 and shit code and whatever. This level of pyramid of doom is just a sore.

Subproc the stack interaction and utilize early return within that subproc, instead of creating >5 levels of indentation.